libbeat/common/cli/confirm.go

```diff
@@ -32,18 +32,18 @@ import (
 // returns true for yes, false for no
 func Confirm(prompt string, def bool) (bool, error) {
 	reader := bufio.NewReader(os.Stdin)
-	return confirm(reader, prompt, def)
+	return confirm(reader, os.Stdout, prompt, def)
 }
 
-func confirm(r io.Reader, prompt string, def bool) (bool, error) {
+func confirm(r io.Reader, out io.Writer, prompt string, def bool) (bool, error) {
```

Comment:
`go test -json` does not always report a test status if the test did write to stdout/stderr directly. I found it easier to modify the code/tests and capture the output as well instead of trying to fix the testing evaluation in gotestsum.

As a nice bonus I modified the tests to actually capture and validate the produced output.